src/libs/sailors/src/sailors.cpp

```diff
@@ -1170,9 +1156,26 @@ uint64_t Sailors::ProcessMessage(MESSAGE &message)
 
         shipID = message.EntityID();
         const std::string &c = message.String();
+        std::vector<std::string> shipManModels;
+
+        if (message.GetFormat() == "lise")
+        {
+            auto *pvd = message.ScriptVariablePointer();
+
+            if (pvd != nullptr)
+            {
+                const int nq = pvd->GetElementsNum();
+                const char *pstr;
+                for (auto i = 0; i < nq; i++)
+                {
+                    pvd->Get(pstr, i);
+                    shipManModels.push_back(pstr);
+                }
+            }
+        }
 
         shipWalk.emplace_back();
-        if (shipWalk[shipsCount].Init(shipID, editorMode, c.c_str()))
+        if (shipWalk[shipsCount].Init(shipID, editorMode, c.c_str(), shipManModels))
```

Comment:
I'd accept shipManModels by value, and move it into Init, rather than passing a reference and killing it from Init

Reply:
copy-elision is not guaranteed here (and it doesn't actually happen). and having a method that accepts only rvalue-ref is as questionable as modifying object by lvalue-ref
but I'd go for it anyway